src/meta/src/manager/catalog/mod.rs

```diff
@@ -920,6 +921,12 @@ impl CatalogManager {
         table_id: TableId,
         internal_table_ids: Vec<TableId>,
     ) -> MetaResult<()> {
+        if self.table_is_created(table_id).await {
```

Comment:
Hmm, in general that makes sense. But the issue is that if stream job created successfully, there will be a catalog update to frontend. So now cancel also needs to synchronize with the frontend catalog.

Let me think of how to resolve it. 🤔

Reply:
Emm, if we issued a cancel command, it means that the stream manager is handling the cancel processing, even the create command is completed, the stream manager will not return `MetaError::cancelled("create".into())` and the catalog will not be synced to frontend.

Reply:
When stream job is finished, catalog will always be synced to frontend.

Consider the following sequence of events:
1. Stream job creating
2. Issued cancel command
3. Stream job created + notified frontend.
4. Cancel command succeeded.

In frontend, when we attempt to create an MV with the same definition, because of step 3, we will receive a `duplicate` error.

Reply:
Actually the step 3 will never occurs. Because create and cancel are both handled by a single thread inside stream manager, and steps 3 and 4 are mutually exclusive. If step 3 success, the cancel command will be ignored since there will be no tracking info in `creating_job_info` for it.

Reply:
It could also be that `barrier manager` is the one notifying though. For recovered stream jobs, barrier manager could be separate thread from the cancel.

Reply:
> If step 3 success, the cancel command will be ignored since there will be no tracking info in creating_job_info for it.

Yeah makes sense. Probably because we didn't cleanup the state previously.

That means this logic is unnecessary. I will revert it.

Reply:
Resolved https://github.com/risingwavelabs/risingwave/pull/13070/commits/2c4537bf5e2125c65d4db9303a08b2e9a8807b32